### PR TITLE
fixed #17866: [bug] EventTouch.getUIDelta return wrong delta values on native platforms

### LINKED ISSUE #17866: [bug] EventTouch.getUIDelta return wrong delta values on native platforms
Reported at https://forum.cocos.org/t/topic/163100

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #17866

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: pal/input/native/touch-input.ts [hunk 1]
```diff
@@ -25,63 +25,132 @@
...
+            const src = changedTouches[i];
+            let dst = cachedTouches[i] as any;
+            if (!dst) {
+                (cachedTouches[i] as any) = dst = {
+                    identifier: 0,
+                    clientX: 0,
+                    clientY: 0,
+                    pageX: 0,
+                    pageY: 0,
+                };
```
Comment (minggo) [inline pal/input/native/touch-input.ts]: Can assign the value with src.

File: pal/input/native/mouse-input.ts [hunk 2]
```diff
@@ -67,53 +127,80 @@ export class MouseInputSource {
...
 
         // Treat window leave/enter events as mouse events as web.
-        jsb.onWindowLeave = this._handleWindowLeave.bind(this);
-        jsb.onWindowEnter = this._handleWindowEnter.bind(this);
+        jsb.onWindowLeave = this._handleWindowLeave;
+        jsb.onWindowEnter = this._handleWindowEnter;
+    }
+
+    private _createEventCacheCallback$ (eventType: InputEventType) {
+        return (mouseEvent?: jsb.MouseEvent | jsb.MouseWheelEvent): void => {
```
Comment (minggo) [inline pal/input/native/mouse-input.ts:138]: Why mouseEvent is optional?

**Comment by dumganhar** [comment]
@cocos-robot run test cases

File: pal/input/native/mouse-input.ts (hunk 2 above)
Comment (dumganhar) [reply pal/input/native/mouse-input.ts:138]: FOR MouseEnter and MouseLeave, they dont have event info and only have type

File: pal/input/native/touch-input.ts (hunk 1 above)
Comment (dumganhar) [reply pal/input/native/touch-input.ts]: Updated.

Use `Object.assign` to copy data from native.

**Review by minggo — APPROVED** [review]

**Pull request merged**